packages/lib/gpc/src/gpcCompile.ts

```diff
@@ -751,27 +770,33 @@ function compileProofEntryConstraints(
     entryEqualToOtherEntryByIndex: CircuitSignal[];
   };
   entryConstraintMetadata: {
-    firstOwnerIndex?: number;
+    firstOwnerIndex: Record<IdentityProtocol, number | undefined>;
   };
 } {
   // Deal with equality comparision and POD ownership, which share circuitry.
-  let firstOwnerIndex = undefined;
+  const firstOwnerIndex = Object.fromEntries(
```

Comment:
This is a style nit, but wouldn't this be clearer and equivalent?
```ts
  const firstOwnerIndex = { SEMAPHORE_V3: undefined, SEMAPHORE_V4: undefined };
```
If there were many owner types defined in an array defined elsewhere, I could see the usefulness of a map(), but given the two constants are used here anyway, the way above seems more direct.